app/Http/Controllers/SignUpController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use DB;
+use App\Http\Requests;
+use Auth;
+use Otp\GoogleAuthenticator;
+
+class SignUpController extends Controller
+{
+    public function signUpPage()
+    {
+        return view('signup');
+    }
+
+    public function signUp(Request $request)
+    {
+        $email = strtolower($request->input('email'));
+        $password1 = $request->input('password1');
+        $password2 = $request->input('password2');
+        if ($password1 != $password2) {
+            return view('signup', ['errorMessage' => 'Passwords do not match.']);
+        } else {
+            $users = DB::table('users')->where('email', $email)->get();
+            if (!empty($users)) {
+                return view('signup', ['errorMessage' => 'That email is already in use']);
+            } else {
+                $id = DB::table('users')->insertGetId(
```

Comment:
Same here, you can create a new user with an array using the `User` model. You have to modify fillable in the App\User class first. https://laravel.com/docs/5.2/eloquent#inserting-and-updating-models